x/ccv/provider/keeper/migration.go

```diff
@@ -0,0 +1,74 @@
+package keeper
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	providertypes "github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+)
+
+// Migrator is a struct for handling in-place store migrations.
+type Migrator struct {
+	ccvProviderKeeper     Keeper
+	ccvProviderParamSpace paramtypes.Subspace
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(ccvProviderKeeper Keeper, ccvProviderParamSpace paramtypes.Subspace,
+) Migrator {
+	return Migrator{ccvProviderKeeper: ccvProviderKeeper, ccvProviderParamSpace: ccvProviderParamSpace}
+}
+
+func (m Migrator) Migratev1Tov2(ctx sdk.Context) error {
```

Comment:
TODO: add method for migration from v1.1.0-multiden to v2, along with v1.2.0-multiden to v2 for consumer